docs/proposals/operator-custom-crd-download-url/README.md

```diff
@@ -0,0 +1,103 @@
+
+The default behavior is sensible and will be retained. However, it should not be the behavior when the intent is to create a new Karmada instance with control plane components of a different version than that of the operator's release version. 
+To support that scenario, we require the ability to explicitly set, as part of the Karmada resource, the URL that should be used to download the CRDs tarball for that Karmada instance.
+
+In addition to making sure required CRDs are downloaded, the Karmada operator also employs a caching strategy to avoid re-downloading CRDs it has already downloaded. It's a cache on the local file system.
+Downloaded CRDs are stored at a well-known location on the local file system based the release version of the Karmada operator. When provisioning a new Karmada instance, before attempting to download the required CRDs, the operator will fist check if
+they already exist in the cache. To support custom CRD download URLs, we require this caching strategy be extended to be more generic.
+
+## Design Details
+
+To support downloading CRDs from a custom URL, we need to add the new fields `CRDRemoteURL` and `CRDDownloadPolicy` to the Karmada resource spec:
+```go
+// CRDDownloadPolicy specifies a policy for how the Karmada operator will retrieve the Karmada CRDs when provisioning a Karmada instance.
+// +enum
+type CRDDownloadPolicy string
+
+const (
+    // DownloadAlways instructs the Karmada operator to always download the CRDs tarball from a remote location.
+    DownloadAlways CRDDownloadPolicy = "Always"
+    // DownloadIfNotPresent instructs the Karmada operator to download the CRDs tarball from a remote location only if it is not yet present in the local cache.
+    DownloadIfNotPresent CRDDownloadPolicy = "IfNotPresent"
+)
+
+// KarmadaSpec is the specification of the desired behavior of the Karmada.
+type KarmadaSpec struct {
+
+HostCluster *HostCluster `json:"hostCluster,omitempty"`
+
+PrivateRegistry *ImageRegistry `json:"privateRegistry,omitempty"`
+
+Components *KarmadaComponents `json:"components,omitempty"`
+
+FeatureGates map[string]bool `json:"featureGates,omitempty"`
+
+// Custom URL that should be used to download required CRDs for this Karmada instance.
+// Omit if the intent is to provision a new Karmada instance with control plane components of the same version as the operator's release version.
+// In that scenario, the CRDs tarball will be downloaded from a release artifact on GitHub that's of the same version as the operator's release version.
+// When that is not the intent, this field should be explicitly set to ensure the right version of the CRDs is installed for this Karmada instance.
+// +optional
+CRDRemoteURL string `json:"crdRemoteURL,omitempty"`
+
+// Policy that should be used by the Karmada operator to retrieve the Karmada CRDs when provisioning this Karmada instance.
+// Defaults to IfNotPresent.
+// +optional
+CRDDownloadPolicy CRDDownloadPolicy `json:"crdDownloadPolicy,omitempty"`
```

Comment:
```suggestion
CRDDownloadPolicy *CRDDownloadPolicy `json:"crdDownloadPolicy,omitempty"`
```
ditto.

Using a pointer type might not be that convenient sometimes, like, we probably need to check whether it is nil. But I'd prefer to follow the API convention.